packages/provider-elasticsearch/src/index.js

```diff
@@ -109,7 +109,20 @@ let ElasticsearchProvider = (config = { request: {} }) => ({
     node._meta.requests.push(metaObj)
     let count = counter.inc()
     debug('(%s) Request: %O\nOptions: %O', count, request, requestOptions)
-    let response = await search(request, requestOptions)
+
+    let response
+    try {
+      response = await search(request, requestOptions)
+    } catch (e) {
+      // Provide information about the error in the standard `cause` property
+      // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#providing_structured_data_as_the_error_cause
+      e.cause = e.meta ?? {}
+      delete e.meta
+      // Remove duplicated information.
+      delete e.cause.meta
```

Comment:
This is a redundant delete and can be removed.

Reply:
It's not. `e.meta` is not `e.cause.meta`.

Reply:
when you set e.cause it clobbers any data and when you delete e.meta it deletes it for cause also  I ran some quick tests to be sure. It is redundant.

Reply:
That's not the shape of `error` from elastic. Here's a more realistic one

```
{ meta: { a: 1, b: 2, meta: { c: 3, d: 4 } } }
```

Reply:
My point is that if you expect meta to have a meta prop then this is fine, but I would not expect that as it makes little sense. If you are solving for that case this is fine as I said above.

Reply:
> My point is that if you expect meta to have a meta prop then this is fine

Yes that is the expectation